File: frontend/src/app/features/character/dto/load-character.dto.ts
import { InventoryItemDto } from '../../../shared/dto/inventory-item.dto';
import { CharacterDto } from './character.dto';
import { EquippedItemDto } from './equipped-item.dto';

export interface LoadCharacterDto {
  character: CharacterDto;
  inventoryItems: InventoryItemDto[];
  equippedItems: EquippedItemDto[];
}

File: backend/Margorak.Api/Dto/LoadCharacterDto.cs
namespace Margorak.Api.Dto
{
    public class LoadCharacterDto
    {
        public CharacterDto Character { get; set; } = null!;
        public List<InventoryItemDto> InventoryItems { get; set; } = [];
        public EquippedItemDto[] EquippedItems { get; set; } = [];
    }
}
